### PR TITLE
pin swapping + Vth (V0) 

### PR DESCRIPTION
**Pull request by openroadie**

### CODE REVIEW

**Review by maliberty — CHANGES_REQUESTED** [review]

**Comment by QuantamHD** [comment]
In the future I think it would be good to break up Resizer.cc into multiple files. It's kind of turning into a giant God file.

File: src/rsz/include/rsz/Resizer.hh [hunk 1]
```diff
@@ -600,6 +607,9 @@ protected:
   int inserted_buffer_count_;
   bool buffer_moved_into_core_;
   // Slack map variables.
+  // This is the minimum length of wire that is worth while to split and
+  // insert a buffer in the middle of. Theoretically computed using the smallest
+  // drive cell (because larger ones would give us a longer length).
   float max_wire_length_;
```
[the remark this answers is not in the document]
Comment (openroadie) [reply src/rsz/include/rsz/Resizer.hh:613]: I can remove the comment if (which reflects my understanding of it when I looked at it last). I would rather not get into the details of this within this PR (if that is ok). LMK

File: src/rsz/src/Resizer.cc [hunk 2]
```diff
@@ -865,6 +866,67 @@ Resizer::findTargetCell(LibertyCell *cell,
   return best_cell;
 }
 
+void
+Resizer::swapPins(Instance *inst, LibertyPort *pin1,
+                  LibertyPort *pin2, bool journal)
+{
+    Pin *found_pin1, *found_pin2;
+    Net *net1, *net2;
+    LibertyPort *port1, *port2;
```
[the remark this answers is not in the document]
Comment (openroadie) [reply src/rsz/src/Resizer.cc:875]: It was really off by 3 lines. Not sure if that makes a whole lot of difference.

[the remark this answers is not in the document]
Comment (maliberty) [reply src/rsz/src/Resizer.cc:875]: Not that big in this case but I mention it so you are aware of the preferred coding style.

File: src/rsz/include/rsz/Resizer.hh (hunk 1 above)
[the remark this answers is not in the document]
Comment (maliberty) [reply src/rsz/include/rsz/Resizer.hh:613]: I can live with it.  I find the whole idea confusing at best.

File: src/rsz/src/Resizer.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (maliberty) [reply src/rsz/src/Resizer.cc:875]: You didn't get the idea it seems.  

874: Net *net1, *net2;
....
886:  net1 = net2 = nullptr;

Just move the definition to 886.  Ditto the others.

File: src/rsz/include/rsz/Resizer.hh (hunk 1 above)
[the remark this answers is not in the document]
Comment (openroadie) [reply src/rsz/include/rsz/Resizer.hh:613]: I will address it in a future PR.  I agree with you.

File: src/rsz/src/Resizer.cc [hunk 3]
```diff
@@ -865,6 +866,67 @@ Resizer::findTargetCell(LibertyCell *cell,
...
+    found_pin1 = found_pin2 = nullptr;
+    net1 = net2 = nullptr;
+    while (pin_iter->hasNext()) {
+        Pin *pin = pin_iter->next();
+        Net *net = network_->net(pin);
+        LibertyPort *port = network_->libertyPort(pin);
+        if (port == pin1) {
+            found_pin1 = pin;
+            net1 = net;
+            port1 = port;
```
[the remark this answers is not in the document]
Comment (openroadie) [reply src/rsz/src/Resizer.cc:894]: Confused myself with variable names. Will fix. Thanks :)

**Review by maliberty — APPROVED** [review]

**Pull request merged**